compendium/ServiceComponent/test/CMakeLists.txt

```diff
@@ -62,6 +67,21 @@ target_link_libraries(${us_servicecomponent_test_exe_name}
         usServiceComponent
 )
 
+foreach(exe_name ${compile_only_exe_names})
+    target_link_libraries(${exe_name} 
+	PRIVATE
+		${GTEST_BOTH_LIBRARIES}
+		${GMOCK_BOTH_LIBRARIES}
+        CppMicroServices
+        usServiceComponent
+    )
+endforeach()
+
+#Excluding compile-only tests from default build to avoid build failure
+foreach(exe_name ${compile_only_exe_names})
+    set_target_properties(${exe_name} PROPERTIES EXCLUDE_FROM_ALL TRUE EXCLUDE_FROM_DEFAULT_BUILD TRUE)
+endforeach()
```

Comment:
there are multiple `foreach` loops in this file which set options/properties on the set of compile time test EXEs. Can the number of `foreach` loops be reduced by combining the operations into 1 `foreach` loop?